src/server.rs

```diff
@@ -135,4 +244,114 @@ mod tests {
 
         assert_eq!((), result)
     }
+
+    #[test]
+    fn test_did_open() {
+        let server = create_server();
+        let params = lsp::DidOpenTextDocumentParams {
+            text_document: lsp::TextDocumentItem::new(
+                lsp::Url::parse("file:///home/user/file.flux")
+                    .unwrap(),
+                "flux".to_string(),
+                1,
+                "from(".to_string(),
+            ),
+        };
+
+        let result = block_on(server.did_open(params));
+
+        assert_eq!((), result);
+    }
+
+    // If the file hasn't been opened on the server get, return an error.
+    #[test]
+    fn test_signature_help_not_opened() {
+        let server = create_server();
+
+        let params = lsp::SignatureHelpParams {
+            context: None,
+            text_document_position_params:
+                lsp::TextDocumentPositionParams::new(
+                    lsp::TextDocumentIdentifier::new(
+                        lsp::Url::parse(
+                            "file:///home/user/file.flux",
+                        )
+                        .unwrap(),
+                    ),
+                    lsp::Position::new(1, 1),
+                ),
+            work_done_progress_params: lsp::WorkDoneProgressParams {
+                work_done_token: None,
+            },
+        };
+
+        let result = block_on(server.signature_help(params));
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_signature_help() {
+        let server = create_server();
+        open_file(&server, "from(".to_string());
+
+        // XXX: rockstar (13 Jul 2021) - In the lsp protocol, Position arguments
+        // are indexed from 1, e.g. there is no line number 0. This references
+        // (0, 5) for compatibility with the previous implementation, but should
+        // be updated to (1, 5) at some point.
+        let params = lsp::SignatureHelpParams {
+            context: None,
+            text_document_position_params:
+                lsp::TextDocumentPositionParams::new(
+                    lsp::TextDocumentIdentifier::new(
+                        lsp::Url::parse(
+                            "file:///home/user/file.flux",
+                        )
+                        .unwrap(),
+                    ),
+                    lsp::Position::new(0, 5),
+                ),
+            work_done_progress_params: lsp::WorkDoneProgressParams {
+                work_done_token: None,
+            },
+        };
+
+        let _expected = Some(lsp::SignatureHelp {
+            signatures: vec![],
+            active_signature: None,
+            active_parameter: None,
+        });
```

Comment:
What's the purpose of declaring this variable that we never use?

Reply:
Ah, thanks. Removed. The compiler is smart enough to drop those, but the intent was to do more than just assert lengths of the returned values. Unfortunately, that test returns 64 signatures, which could make a big unwieldy test. I wanted to stay as close to the original server's tests as possible, for now. In the future, a better flux script would make this more manageable.